### ISSUE TITLE
Leverage Jekyll for final output

### ISSUE THREAD

**Issue by benbalter** [theodi/octopub#7]
Awesome stuff! :metal:

Reading [the announcement post](http://theodi.org/blog/removing-barriers-to-publishing-open-data), one thing that struck me, the project seems to be reinventing the wheel a bit, and can knock out some open issues (e.g., https://github.com/theodi/git-data-publisher/issues/6 and https://github.com/theodi/git-data-publisher/issues/4), by leveraging Jekyll's built in data capabilities.

> It also generates an HTML representation of the dataset (with DCAT metadata embedded inside), which is accessed via GitHub pages 

I don't know the full context, but if I were to publish data via GitHub, keeping the same _exact_ end-user-facing result, here's how I'd do it:
1. Store dataset metadata in `_config.yml`. This way, project metadata is human readable without additional software, easily editable by non-developers, and is more easily diffed if changes are proposed via pull request.
2. Have Jekyll generate the existing `datapackage.json` file via a liquid template, pulling in the human-readable information from `_config.yml`. Again, the output will be identical to an end user.
3. Store each dataset in the `_data` folder as a `.csv` file. Jekyll will automatically read each file in, and expose it via the `site.data` namespace. 
4. Have Jekyll generate the `index.html` file via a liquid template (pulling from the `_data` files). Again, the final output would remain identical. The big advantage here, is as the data changes (e.g., collaboratively via pull request and issues) the output is automatically kept up to date, without the need to re-run the original software.
5. Expose JSON representations of each file (as simple as `{{ site.data.[FILENAME] | jsonify }}`)

For non-GitHub use cases, you could even package `jekyll` within the app and simply run `jekyll build` as an option before publishing (again, resulting in identical output as a fallback). 

TL;DR: Jekyll will allow you to save human-readable source data, not machine-readable rendered output, and can empower publishers and consumers to adopt a more open-source workflow.

**Comment by pezholio** [comment]
@benbalter - I'm having issues with random Jekyll page build failures when repos are created (see #42). If I pull down a created repo, make a trivial change and push up, the page builds fine. Is this something to do with using the api to push files up? Does the post-receive hook not fire at the Github end?

I'm toying with the idea of doing the Jekyll compilation this end before pushing up, but this seems inelegant, so any help would be appreciated!

**Comment by benbalter** [comment]
It should work. If you email support@github.com, we can look into it.
